app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -913,49 +797,17 @@ private void sendStatus(String content, Status.Visibility visibility, boolean se
             mediaIds.add(item.id);
         }
 
-        Callback<Status> callback = new Callback<Status>() {
-            @Override
-            public void onResponse(@NonNull Call<Status> call, @NonNull Response<Status> response) {
-                if (response.isSuccessful()) {
-                    onSendSuccess();
-                } else {
-                    onSendFailure(response);
-                }
-            }
-
-            @Override
-            public void onFailure(@NonNull Call<Status> call, @NonNull Throwable t) {
-                onSendFailure(null);
-            }
-        };
-        mastodonApi.createStatus(content, inReplyToId, spoilerText, visibility.serverString(),
-                sensitive, mediaIds).enqueue(callback);
-    }
+        Intent sendIntent = SendTootService.sendTootIntent(this, content, spoilerText, inReplyToId,
```

Comment:
If you're unhappy about too many parameters, maybe we should construct something on this side and pass it?